google/resource_compute_instance_group_manager.go

```diff
@@ -212,36 +230,46 @@ func resourceComputeInstanceGroupManagerRead(d *schema.ResourceData, meta interf
 		return err
 	}
 
-	getInstanceGroupManager := func(zone string) (interface{}, error) {
-		return config.clientCompute.InstanceGroupManagers.Get(project, zone, d.Id()).Do()
-	}
-
 	var manager *compute.InstanceGroupManager
-	var e error
-	if zone, ok := d.GetOk("zone"); ok {
-		manager, e = config.clientCompute.InstanceGroupManagers.Get(project, zone.(string), d.Id()).Do()
-
-		if e != nil {
-			return handleNotFoundError(e, d, fmt.Sprintf("Instance Group Manager %q", d.Get("name").(string)))
+	switch computeApiVersion {
+	case v1:
+		getInstanceGroupManager := func(zone string) (interface{}, error) {
+			return config.clientCompute.InstanceGroupManagers.Get(project, zone, d.Id()).Do()
 		}
-	} else {
-		// If the resource was imported, the only info we have is the ID. Try to find the resource
-		// by searching in the region of the project.
-		var resource interface{}
-		resource, e = getZonalResourceFromRegion(getInstanceGroupManager, region, config.clientCompute, project)
-
-		if e != nil {
-			return e
+
+		var v1Manager *compute.InstanceGroupManager
+		var e error
+		if zone, ok := d.GetOk("zone"); ok {
+			v1Manager, e = config.clientCompute.InstanceGroupManagers.Get(project, zone.(string), d.Id()).Do()
+
+			if e != nil {
+				return handleNotFoundError(e, d, fmt.Sprintf("Instance Group Manager %q", d.Get("name").(string)))
+			}
+		} else {
+			// If the resource was imported, the only info we have is the ID. Try to find the resource
+			// by searching in the region of the project.
+			var resource interface{}
+			resource, e = getZonalResourceFromRegion(getInstanceGroupManager, region, config.clientCompute, project)
+
+			if e != nil {
+				return e
+			}
+
+			v1Manager = resource.(*compute.InstanceGroupManager)
 		}
 
-		manager = resource.(*compute.InstanceGroupManager)
-	}
+		if v1Manager == nil {
+			log.Printf("[WARN] Removing Instance Group Manager %q because it's gone", d.Get("name").(string))
+
+			// The resource doesn't exist anymore
+			d.SetId("")
+			return nil
+		}
 
-	if manager == nil {
-		log.Printf("[WARN] Removing Instance Group Manager %q because it's gone", d.Get("name").(string))
-		// The resource doesn't exist anymore
-		d.SetId("")
-		return nil
+		manager, err = convertInstanceGroupManagerToV1(v1Manager)
```

Comment:
This is a bit weird because there is still only a single version - `v1` is the canonical version right now, but won't be once there is another version supported. Removed until necessary.